### PR TITLE
[DeviceSanitizer] Add tests to check customizing red zone size

### PR DESCRIPTION
**Pull request by zhaomaosu**

### CODE REVIEW

File: sycl/test-e2e/AddressSanitizer/common/config-red-zone-size.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// REQUIRES: linux
+// RUN: %{build} %device_sanitizer_flags -O0 -g -o %t
```
Comment (AllanZyne) [inline sycl/test-e2e/AddressSanitizer/common/config-red-zone-size.cpp:2]:
```suggestion
// REQUIRES: linux, cpu
// RUN: %{build} %device_asan_flags -O0 -g -o %t
```

**Comment by kbenzie** [comment]
@zhaomaosu I'd appriciate if you could open this up for review so approvals are in place before we merge the UR changes.

**Review by yingcong-wu — APPROVED** [review]
lgtm

**Comment by kbenzie** [comment]
Are these [test failures](https://github.com/intel/llvm/actions/runs/9077514283/job/24943496121?pr=13505) due to a dependency on the SYCL RT changes in https://github.com/intel/llvm/pull/13503?

**Comment by zhaomaosu** [comment]
> Are these [test failures](https://github.com/intel/llvm/actions/runs/9077514283/job/24943496121?pr=13505) due to a dependency on the SYCL RT changes in #13503?

Yeah, I'll re-run the pre-ci after #13503 is merged.

**Comment by kbenzie** [comment]
> > Are these [test failures](https://github.com/intel/llvm/actions/runs/9077514283/job/24943496121?pr=13505) due to a dependency on the SYCL RT changes in #13503?
> 
> Yeah, I'll re-run the pre-ci after #13503 is merged.

This has been merged now.

**Review by bso-intel — APPROVED** [review]

**Pull request merged**